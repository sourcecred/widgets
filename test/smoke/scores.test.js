"use strict";

const minScoresV1 = require("./data/min-scores-v1.json");
const minScoresV2 = require("./data/min-scores-v2.json");
const {createTmpCacheDir, clearTmpCacheDir} = require("../util/caching");
const {scoreService} = require("../../lib");

test("Expect min-scores-v1 to validate", async () => {
  // Given

  // When
  const credData = scoreService.validateScoreFormat(minScoresV1);

  // Then
  expect(credData).toMatchSnapshot();
});

test("Expect min-scores-v2 to validate", async () => {
  // Given

  // When
  const credData = scoreService.validateScoreFormat(minScoresV2);

  // Then
  expect(credData).toMatchSnapshot();
});

test("Expect min-scores-v1 and v2 to normalize to the same output", async () => {
  // Given

  // When
  const v1Data = scoreService.validateScoreFormat(minScoresV1);
  const v2Data = scoreService.validateScoreFormat(minScoresV2);

  // Then
  expect(v1Data).toEqual(v2Data);
});
