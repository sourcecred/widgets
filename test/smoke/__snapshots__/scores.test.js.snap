// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect min-scores-v1 to validate 1`] = `
Object {
  "intervals": Array [
    Object {
      "endTimeMs": 1540080000000,
      "startTimeMs": 1539475200000,
    },
  ],
  "users": Array [
    Object {
      "address": Array [
        "sourcecred",
        "github",
        "USERLIKE",
        "USER",
        "credbot",
      ],
      "intervalCred": Array [
        42.123,
      ],
      "totalCred": 42.123,
    },
  ],
}
`;

exports[`Expect min-scores-v2 to validate 1`] = `
Object {
  "intervals": Array [
    Object {
      "endTimeMs": 1540080000000,
      "startTimeMs": 1539475200000,
    },
  ],
  "users": Array [
    Object {
      "address": Array [
        "sourcecred",
        "github",
        "USERLIKE",
        "USER",
        "credbot",
      ],
      "intervalCred": Array [
        42.123,
      ],
      "totalCred": 42.123,
    },
  ],
}
`;
